python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,258 @@
+import unittest
+import warnings
+import pandas as pd
+import numpy as np
+from sklearn import datasets
+from sklearn.linear_model import Perceptron, LogisticRegression
+from sklearn.utils import check_random_state
```

Comment:
One main point in doing these sanity tests is to be independent of ml4ir in the sense that I cannot use ml4ir to test ml4ir.

Reply:
I can do that but that would make the tests essentially frozen to a specific number of tests with specific datasets. The way I had in mind to make it configurable so that we can add easily more tests and control the difficulties.

Reply:
I kinda agree with @mohazahran here. If we don't have `sklearn`, and want to compare with another model trained on `ml4ir` we might not be seeing the full suite of issues. But I think what @jakemannix proposes is a good middle ground for this.